contributors/design-proposals/network/service_port_ranges.md

```diff
@@ -0,0 +1,128 @@
+# Support port ranges in services
+
+Authors: @m1093782566
+
+## Background & Motivations
+
+There are several applications like SIP apps or RTP which needs a lot of ports to run multiple calls or media streams. Currently there is no way to allow a range in ports in spec. So essentially users may have to do this:
+
+```
+- name: sip-udp5060
+  containerPort: 5060
+  protocol: UDP
+- name: sip-udp5061
+  containerPort: 5061
+  protocol: UDP
+  ...
+- name: sip-udp5559
+  containerPort: 5559
+  protocol: UDP
+```
+
+Doing above for 500 ports is not pretty. Can we have a way to allow port ranges like 5060-5559?
+
+## Objective
+
+This proposal builds off of earlier requests to [Support port ranges in services](https://github.com/kubernetes/kubernetes/issues/23864) and aims to document all the requirements of supporting port ranges in services. Each port range is a single port or single contiguous port range, ranging from low to high, for which this forwarding rule matches. Packets of the specified protocol sent to these ports will be forwarded on to the appropriate target instances.
+
+## Goals
+
+A server side solution to support single or multiple contiguous port range.
+
+## Non-goal
+
+Target port remapping in ranges.
+
+Pod port ranges.
+
+## User Cases
+
+* Specifying port ranges in services declarations is mandatory for any VoIP-related application.
+* With the standard RTP port range which is 16384-32767. Clients connect on the standard RTP port range with a media server application.
+* Needs this change to make TeleStax (an RTC framework) work well in Kubernetes.
+
+## Proposal
+
+This proposal builds off earlier feature requrest to [Support port ranges in services](https://github.com/kubernetes/kubernetes/issues/23864). We aims to implement service port ranges at least for iptables and IPVS modes.
+
+### Service API changes
+
+Create a new filed `portRangeSize` in `ServicePort`.
```

Comment:
nit: field

Reply:
Thanks for comment, I will create a new KEP. You can continue to put comments there.